libs/martech/analytics.js

```diff
@@ -0,0 +1,48 @@
+export function processTrackingLabels(text, charLimit = 20) {
+  return text?.trim().replace(/\s+/g, ' ').split('|').join(' ')
+    .slice(0, charLimit);
+}
+
+export function decorateDefaultLinkAnalytics(block) {
+  if (block.classList.length
+    && !block.className.includes('metadata')
+    && !block.classList.contains('link-block')
+    && !block.classList.contains('section')
+    && block.nodeName === 'DIV') {
+    let header = '';
+    let linkCount = 1;
+    block.querySelectorAll('h1, h2, h3, h4, h5, h6, a:not(.video.link-block), button, .heading-title')
+      .forEach((item) => {
+        if (item.nodeName === 'A' || item.nodeName === 'BUTTON') {
+          if (!item.hasAttribute('daa-ll')) {
+            let label = item.textContent;
+            if (label.trim() === '') {
```

Comment:
Why not trim it directly when defining the `label` above?

Reply:
moved